src/patternfly/components/Tabs/tabs.scss

```diff
@@ -1,104 +1,118 @@
+  --pf-c-tabs--m-vertical--c-tabs__scroll-button--PaddingBottom: calc(var(--pf-global--spacer--xl) / 2);
+
+  // Vertical box
+  --pf-c-tabs--m-vertical--m-box--Inset: var(--pf-global--spacer--xl);
+
+  @media screen and (min-width: $pf-global--breakpoint--md) {
+    // Button
+    --pf-c-tabs__button--PaddingTop: var(--pf-c-tabs__button--md--PaddingTop);
+    --pf-c-tabs__button--PaddingBottom: var(--pf-c-tabs__button--md--PaddingBottom);
+    --pf-c-tabs--m-roomy--c-tabs__button--PaddingRight: var(--pf-c-tabs--m-roomy--c-tabs__button--md--PaddingRight);
+    --pf-c-tabs--m-roomy--c-tabs__button--PaddingLeft: var(--pf-c-tabs--m-roomy--c-tabs__button--md--PaddingLeft);
+
+    // Scroll button
+    --pf-c-tabs__scroll-button--Width: var(--pf-c-tabs__scroll-button--md--Width);
+  }
 
   position: relative;
   display: flex;
-  flex-direction: column;
-  align-items: flex-start;
-
-  // Scroll buttons hidden by default
-  .pf-c-tabs__scroll-button {
-    display: none;
-    visibility: hidden;
-  }
-
-  // Scroll buttons shown with modifiers
-  &.pf-m-start .pf-c-tabs__scroll-button:nth-of-type(1),
-  &.pf-m-end .pf-c-tabs__scroll-button:nth-of-type(2) {
-    display: block;
-    visibility: visible;
-  }
+  padding-right: var(--pf-c-tabs--Inset);
+  padding-left: var(--pf-c-tabs--Inset);
+  overflow: hidden;
 
-  // Scroll button current state
-  &.pf-m-start-current:not(.pf-m-tabs-secondary) .pf-c-tabs__scroll-button:nth-of-type(1),
-  &.pf-m-end-current:not(.pf-m-tabs-secondary) .pf-c-tabs__scroll-button:nth-of-type(2) {
-    &::after {
-      position: absolute;
-      top: 0;
-      right: 0;
-      bottom: 0;
-      left: 0;
-      margin-left: var(--pf-c-tabs__item--BorderWidth);
-      content: "";
-      border-top: var(--pf-c-tabs__item--m-current--BorderTopWidth) solid var(--pf-c-tabs__item--m-current--Color);
-    }
+  &::before {
+    position: absolute;
+    right: 0;
+    bottom: 0;
+    left: 0;
+    width: 100%;
+    content: "";
+    border-bottom: var(--pf-c-tabs--before--BorderWidth) solid var(--pf-c-tabs--before--BorderColor);
   }
 
-  // Scroll button current state
-  &.pf-m-start-current.pf-m-tabs-secondary .pf-c-tabs__scroll-button:nth-of-type(1),
-  &.pf-m-end-current.pf-m-tabs-secondary .pf-c-tabs__scroll-button:nth-of-type(2) {
-    color: var(--pf-c-tabs__scroll-button--m-secondary-right--m-start-current--Color);
-  }
-
-  // Primary with secondary tabs style
-  + &.pf-m-tabs-secondary {
-    margin-top: calc(-1 * var(--pf-c-tabs__item--BorderWidth));
-    border-top: var(--pf-c-tabs__item--BorderWidth) solid var(--pf-c-tabs__item--BorderColor);
-
-    .pf-c-tabs__scroll-button {
-      margin-top: calc(-1 * var(--pf-c-tabs__item--BorderWidth));
-    }
+  &.pf-m-secondary::before,
+  &.pf-m-no-border::before {
+    content: none;
   }
 
   // Filled style
   &.pf-m-fill {
     .pf-c-tabs__list {
-      width: 100%;
+      flex-basis: 100%;
     }
 
     .pf-c-tabs__item {
       flex: 1;
```

Comment:
If you set this to `flex-grow: 1;`, I think you can remove `white-space: nowrap;`